### PR TITLE
feat(native plugin): add sql source native plugin

### PR DESCRIPTION
**Pull request by SuperRxan**
Signed-off-by: Jianxiang Ran <rxan_embedded@163.com>

### CODE REVIEW

**Comment by ngjaying** [comment]
commit name typo fea

File: extensions/sources/sql/sql.go [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+			return
+		}
+	}
+}
+
+func (m *sqlsource) Close(ctx api.StreamContext) error {
+	logger := ctx.GetLogger()
+	logger.Debugf("Closing sql stream to %v", m.conf)
+
+	if m.db != nil {
```
Comment (ngjaying) [inline extensions/sources/sql/sql.go:131]: put this in Open by defer

File: extensions/sqldatabase/driver/commonSql.go [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+	}
+	return in
+}
+
+func (q *QueryGenerator) SqlQueryStatement() (string, error) {
+	return q.getSelect() + q.getCondition() + q.getOrderby() + q.getLimit(), nil
+}
+
+func (q *QueryGenerator) CollectIndexValue(row map[string]interface{}) {
+	if q.indexSlice == nil {
```
Comment (ngjaying) [inline extensions/sqldatabase/driver/commonSql.go:76]: Is this needed? Why not just get the row's last value

File: extensions/sources/sql/sql.go [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+				if err != nil {
+					logger.Errorf("Run sql scan(%s) error %v", query, err)
+					errCh <- err
+					return
+				}
+
+				for i, colName := range cols {
+					data[colName] = columns[i]
+				}
+				m.Query.CollectIndexValue(data)
```
Comment (ngjaying) [inline extensions/sources/sql/sql.go:117]: looks like over wrapping for me. Maybe just count the max value here

File: extensions/sqldatabase/driver/mysql/entry.go [hunk 4]
```diff
@@ -0,0 +1,10 @@
+package sqlserver
+
+import (
+	_ "github.com/go-sql-driver/mysql"
```
Comment (ngjaying) [inline extensions/sqldatabase/driver/mysql/entry.go:4]: Add built tag for all drivers

File: extensions/sources/sql/sql.go (hunk 1 above)
Comment (SuperRxan) [reply extensions/sources/sql/sql.go:166]: ack

File: extensions/sqldatabase/driver/commonSql.go (hunk 2 above)
Comment (SuperRxan) [reply extensions/sqldatabase/driver/commonSql.go]: ack

File: extensions/sources/sql/sql.go (hunk 3 above)
Comment (SuperRxan) [reply extensions/sources/sql/sql.go]: ack

**Review by ngjaying — APPROVED** [review]

**Review by ngjaying — APPROVED** [review]

**Pull request merged**